setup.py

```diff
@@ -1124,6 +1124,7 @@ def main():
         "networkx",
         "jinja2",
         "fsspec",
+        "pynvml>=11.4.1",
```

Comment:
Adding such a hard dependency is definitely not ok without much deeper considerations.